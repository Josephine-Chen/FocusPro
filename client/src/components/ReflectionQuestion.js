import React from 'react';
import { Row, Input, Button, Table } from 'react-materialize';
import { getExtensionDataAjax, getBlacklistAjax } from '../utils/SettingsUtil';

export default class ReflectionQuestion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(),
      extensionData: [],
    };
  }

  componentDidMount() {
    this.getBlacklist();
  }

  getBlacklist() {
    getBlacklistAjax(this.state.profile.user_id, (data) => {
      this.getExtensionData(data.data);
    });
  }

  getExtensionData(blacklistData) {
    const exDataArr = [];
    blacklistData.forEach((blacklist) => {
      getExtensionDataAjax(blacklist, (exData) => {
        exDataArr.push(exData);
      });
    });
    this.setState({
      extensionData: exDataArr,
    });
  }

  render() {
    return (
      <div>
        {console.log('load blacklist func?', getBlacklistAjax)}
        <h1>Awareness Questions </h1>
        <h5>How much time have you been spending on the blacklist sites?</h5>
        <AwarenessQuestionTable exData={this.state.extensionData} />
        <Button>Answer Awareness Questions</Button>
        <br />
        <br />
        <Row>
          <h1>Reflection Questions</h1>
          <h5>Not bad, you are aware of how much time you lost on the distraction sites! Please think about:</h5>
        </Row>
        <Row>
          <div>What is already working in your system that you can build on?</div>
          <Input s={12}></Input>
        </Row>
        <Row>
          <div>What might 'help' look like?</div>
          <Input s={12}></Input>
        </Row>
      </div>
    );
  }


}

const AwarenessQuestionTable = ({ exData }) => (
  <Table>
    <thead>
      <tr>
        <th data-field="sites">Blacklist sites</th>
        <th data-field="guessedTime">Time you think you spent</th>
        <th data-field="actualTime">Actual time spent (min)</th>
      </tr>
    </thead>
    <tbody>
      {exData.map(data => (
        <tr>
          <td>{data.url}</td>
          <td><Input type="select">
            <option value="0"></option>
            <option value="1">0~30min</option>
            <option value="2">30min~1hr</option>
            <option value="3">1hr~2hr</option>
            <option value="4">2hr+</option>
          </Input></td>
          <td>{data.time_spent}</td>
        </tr>
      ))}
    </tbody>
  </Table>
);
