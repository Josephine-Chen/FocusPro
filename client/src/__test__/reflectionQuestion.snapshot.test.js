import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../utils/SettingsUtil', () => {
  let date = (new Date()).setHours(0,0,0,0);
  const dummyQues ={
    data: [{
            question: 'What help can you get?',
            answer: 'Get a couch to hold myself accountable',
            updatedAt: date,
          }, {
            question: 'Whats the worst thing that could happen?',
            answer: 'nothing',
            updatedAt: new Date(date.valueOf() - 1000*60*60*24),
    }],
  };
  const dummyBlacklist ={
    data: [
      {
        url:'facebook.com',
        blacklist_type:1,
        blacklist_time:3,
        updatedAt:'2017-03-08T03:23:46.663Z',
      }
    ]
  };

  const dummyExtension = {
    url: 'facebook.com',
    time_spent: 30,
  };

  return {
    getReflectionsAjax: (userId, callback) => {
      callback(dummyQues);
    },

    getBlacklistAjax: (userId, callback) =>{
      callback(dummyBlacklist);
    },
    getExtensionDataAjax: (blacklist, callback) =>{
      callback(dummyExtension);
    }
  }
});

import ReflectionQuestion from '../components/ReflectionQuestion.js';


describe('ReflectionQuestion (Snapshot)', () => {
  beforeAll(() => {
    global.auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  afterAll(() => {
    global.auth = null;
  });

  it('ReflectionQuestion renders', () => {
    const component = renderer.create(<ReflectionQuestion auth={auth}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});