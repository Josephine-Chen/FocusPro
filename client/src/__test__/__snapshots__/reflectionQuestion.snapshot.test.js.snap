exports[`ReflectionQuestion (Snapshot) ReflectionQuestion renders 1`] = `
<div>
  <h1>
    Awareness Questions 
  </h1>
  <h5>
    How much time have you been spending on the blacklist sites?
  </h5>
  <table
    className="">
    <thead>
      <tr>
        <th
          data-field="sites">
          Blacklist sites
        </th>
        <th
          data-field="guessedTime">
          Time you think you spent
        </th>
        <th
          data-field="actualTime">
          Actual time spent (min)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          facebook.com
        </td>
        <td>
          <div
            className="col input-field">
            <select
              className=""
              defaultValue={undefined}
              id="input_0"
              multiple={undefined}
              onChange={[Function]}>
              <option
                value="0" />
              <option
                value="1">
                0~30min
              </option>
              <option
                value="2">
                30min~1hr
              </option>
              <option
                value="3">
                1hr~2hr
              </option>
              <option
                value="4">
                2hr+
              </option>
            </select>
          </div>
        </td>
        <td
          className={undefined}
          style={
            Object {
              "display": "none",
            }
          }>
          30
        </td>
      </tr>
    </tbody>
  </table>
  <button
    className="btn"
    disabled={false}
    onClick={[Function]}>
    Answer Awareness Questions
  </button>
  <br />
  <br />
  <div />
</div>
`;
